climpred/classes.py

```diff
@@ -152,6 +153,113 @@ def __repr__(self):
         else:
             return _display_metadata(self)
 
+    def _math(self, other, operator):
+        """Helper function for __add__, __sub__, __mul__, __truediv__.
+
+        Allows math operations with type:
+            - int
+            - float
+            - np.ndarray
+            - xr.DataArray without new dimensions
+            - xr.Dataset without new dimensions or variables
+
+        """
+        assert isinstance(operator, str)
+
+        def add(a, b):
+            return a + b
+
+        def sub(a, b):
+            return a - b
+
+        def mul(a, b):
+            return a * b
+
+        def div(a, b):
+            return a / b
+
+        ALLOWED_TYPES_FOR_MATH_OPERATORS = [
+            int,
+            float,
+            np.ndarray,
+            xr.DataArray,
+            xr.Dataset,
+            type(self),
+        ]
+        # catch other dimensions in other
+        error_str = f'Cannot use {type(self)} {operator} {type(other)}'
+        if isinstance(other, tuple([xr.Dataset, xr.DataArray])):
+            all_dims = set(
+                [d for d in list(self._datasets['initialized'].dims)] + CLIMPRED_DIMS
```

Comment:
What is the reasoning behind just checking against `self._datasets`? Maybe we could fix the issue discussed in the comments by only applying `other` to a sub-dataset (one in the `self._datasets` dictionary) if its dimensions are a subset of the given `self._datasets`.

E.g.

```python
if set(other_datasets.dims).issubset(dataset.dims):
    # apply operation
```

This could would of course go elsewhere.

Reply:
Oh wait no this still breaks for the case that `.sel(time='...')` returns the identical initialized object. I still think we can check the `.issubset()` thing at each dataset instead of only checking initialized here. Seems like this could introduce a bug.

Reply:
reasoning for only checking `initialized`: this is the xr.Dataset that instantiates the object. other xr.Datasets get already checked for DimensionErrors when added.

Reply:
That's a good point. I guess all other objects by definition should be the same dimensions. I agree with this now.